src/SignalR/clients/ts/signalr/src/IHubProtocol.ts

```diff
@@ -91,6 +97,18 @@ export interface StreamItemMessage extends HubInvocationMessage {
     readonly item?: any;
 }
 
+/** A hub message representing a single stream item, transferred through a streaming parameter. */
+export interface StreamDataMessage extends HubMessageBase {
+    /** @inheritDoc */
+    readonly type: MessageType.StreamData;
+
+    /** The streamId. */
+    readonly streamId: string;
+
+    /** The item produced by the client. */
+    readonly item?: any;
+}
```

Comment:
Considering that StreamDataMessage and StreamItemMessage are conceptually and structurally identical, I wonder if it would be better to combine these concepts. You could just make streamIds in server-to-client StreamDataMessages the same as the invocationId. That or assign a unique streamId for response streams even though there can't be more than one per invocation. At least no yet.

Reply:
Yeah, we've talked about this before and have an issue filed about it https://github.com/aspnet/SignalR/issues/2592

@mikaelm12 Also brought it up again this week. We can probably merge this with the current design and open discussions about designing it differently on that issue/in-person.